pkg/component/auditlog-forwarder/deploy_test.go

```diff
@@ -0,0 +1,727 @@
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package auditlogforwarder_test
+
+import (
+	"context"
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/types"
+	appsv1 "k8s.io/api/apps/v1"
+	autoscalingv1 "k8s.io/api/autoscaling/v1"
+	corev1 "k8s.io/api/core/v1"
+	policyv1 "k8s.io/api/policy/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer/json"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	vpaautoscalingv1 "k8s.io/autoscaler/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1"
+	clientcmdlatest "k8s.io/client-go/tools/clientcmd/api/latest"
+	clientcmdv1 "k8s.io/client-go/tools/clientcmd/api/v1"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	fakeclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	forwarderconfigv1alpha1 "github.com/gardener/auditlog-forwarder/pkg/apis/config/v1alpha1"
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	resourcesv1alpha1 "github.com/gardener/gardener/pkg/apis/resources/v1alpha1"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/component"
+	"github.com/gardener/gardener/pkg/resourcemanager/controller/garbagecollector/references"
+	kubernetesutils "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"github.com/gardener/gardener/pkg/utils/retry"
+	retryfake "github.com/gardener/gardener/pkg/utils/retry/fake"
+	secretsmanager "github.com/gardener/gardener/pkg/utils/secrets/manager"
+	fakesecretsmanager "github.com/gardener/gardener/pkg/utils/secrets/manager/fake"
+	"github.com/gardener/gardener/pkg/utils/test"
+	. "github.com/gardener/gardener/pkg/utils/test/matchers"
+
+	auditlogforwarder "github.com/gardener/gardener-extension-auditing/pkg/component/auditlog-forwarder"
+)
+
+var _ = Describe("AuditlogForwarder", func() {
+	var (
+		ctx context.Context
+
+		managedResourceName = "extension-auditing"
+		namespace           = "some-namespace"
+		image               = "europe-docker.pkg.dev/gardener-project/releases/gardener/auditlog-forwarder:v1.0.0"
+
+		fakeClient        client.Client
+		fakeSecretManager secretsmanager.Interface
+		deployer          component.DeployWaiter
+		values            auditlogforwarder.Values
+
+		fakeOps   *retryfake.Ops
+		consistOf func(...client.Object) types.GomegaMatcher
+
+		managedResource       *resourcesv1alpha1.ManagedResource
+		managedResourceSecret *corev1.Secret
+
+		deployment          *appsv1.Deployment
+		service             *corev1.Service
+		podDisruptionBudget *policyv1.PodDisruptionBudget
+		vpa                 *vpaautoscalingv1.VerticalPodAutoscaler
+		serviceAccount      *corev1.ServiceAccount
+		configMap           *corev1.ConfigMap
+		kubeconfigSecret    *corev1.Secret
+		httpOutputSecret    *corev1.Secret
```

Comment:
These resources should have name with prefix `expected`, e.g. `expectedServiceAccount`.